src/main/app.ts

```diff
@@ -51,22 +61,24 @@ if (config.useCSRFProtection === true) {
     },
   };
 
-  app.use(csrf(csrfOptions), (req, res, next) => {
+  app.use(unless("/import", csrf(csrfOptions), (req, res, next) => {
     res.locals.csrfToken = req.csrfToken();
     next();
-  });
+  }));
 }
 
+app.use(unless("/oauth2redirect", authCheckerUserOnlyFilter));
+app.use(unless("/oauth2redirect", serviceFilter));
 app.use("/", RouterFinder.findAll(path.join(__dirname, "routes")));
 
 // returning "not found" page for requests with paths not resolved by the router
-app.use((req, res, next) => {
+app.use((req, res, next) => { // eslint-disable-line no-unused-vars
   res.status(404);
   res.render("not-found");
 });
 
 // error handler
-app.use((err, req, res, next) => {
+app.use((err, req, res, next) => { // eslint-disable-line no-unused-vars
```

Comment:
I think if you aren't using next here but am using res, you only need to do the params up to that point i.e. (err, req, res). Then you don't need the eslint-disable I believe.

Reply:
Not quite correct... I did try this out but it didn't work. Express requires you explicitly pass _four_ arguments for this middleware to be treated as an error handler. See https://expressjs.com/en/guide/error-handling.html